lib/src/models/chat_user.dart

```diff
@@ -67,15 +67,32 @@ class ChatUser {
         id: json["id"],
         name: json["name"],
         profilePhoto: json["profilePhoto"],
-        imageType: json["imageType"],
+        imageType: ImageType.tryParse(json['imageType']?.toString()) ??
+            ImageType.network,
         defaultAvatarImage: json["defaultAvatarImage"],
       );
 
   Map<String, dynamic> toJson() => {
         'id': id,
         'name': name,
         'profilePhoto': profilePhoto,
-        'imageType': imageType,
+        'imageType': imageType.name,
         'defaultAvatarImage': defaultAvatarImage,
       };
+
+  ChatUser copyWith({
+    String? id,
+    String? name,
+    String? profilePhoto,
+    ImageType? imageType,
+    String? defaultAvatarImage,
+  }) {
```

Comment:
Please also add a boolean that would decide if the user wants to override value with null.